test_quality_of_service/test/qos_test_publisher.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "test_quality_of_service/qos_test_publisher.hpp"
+#include "test_quality_of_service/qos_test_node.hpp"
+
+QosTestPublisher::QosTestPublisher(
+  const std::string & name,
+  const std::string & topic,
+  const rclcpp::PublisherOptions<> & publisher_options,
+  const std::chrono::milliseconds & publish_period)
+: QosTestNode(name, topic),
+  publisher_options_(publisher_options),
+  publish_period_(publish_period),
+  timer_(nullptr),
+  is_publishing(false)
+{
+  RCLCPP_INFO(this->get_logger(), "created publisher %s %s\n", name.c_str(), topic.c_str());
+  publisher_ = this->create_publisher<std_msgs::msg::String>(topic_, publisher_options_);
+}
+
+QosTestPublisher::~QosTestPublisher()
```

Comment:
You can remove and add `~QosTestPublisher() = default;`

Reply:
Implemented in latest changes.